### PR TITLE
In-memory engine: evict range should consider loading range

### LINKED ISSUE #17153: In-memory Engine: loading range should be considered in eviction
## Bug Report

loading range should be considered in eviction

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Close #17153

What's Changed:

```commit-message
evict range should consider loading range
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/region_cache_memory_engine/tests/failpoints/test_memory_engine.rs [hunk 1]
```diff
@@ -205,3 +210,66 @@ fn test_clean_up_tombstone() {
...
+        .recv_timeout(Duration::from_secs(5))
+        .unwrap();
+
+    engine.snapshot(range1, 100, 100).unwrap_err();
+    engine.snapshot(range2, 100, 100).unwrap_err();
+    engine.snapshot(range3, 100, 100).unwrap();
+
+    drop(engine);
+
+    std::thread::sleep(Duration::from_secs(1));
```
Comment (LykxSassinator) [inline components/region_cache_memory_engine/tests/failpoints/test_memory_engine.rs:274]: Is the drop with sleep necessary here ? Without it, will this case be flaky ?

Comment (SpadeA-Tang) [reply components/region_cache_memory_engine/tests/failpoints/test_memory_engine.rs:274]: I remember I have deleted this code... I will fix it.

**Review by LykxSassinator — APPROVED** [review]
LGTM

**Review by overvenus — APPROVED** [review]
LGTM

**Pull request merged**